includes/payments/functions.php

```diff
@@ -1218,27 +1221,40 @@ function give_donation_amount( $donation, $format_args = false ) {
 		array(
 			'currency' => false,
 			'amount'   => false,
+
+			// Define context of donation amount, by default keep $type as blank.
+			// Pass as 'stats' to calculate donation report on basis of base amount for the Currency-Switcher Add-on.
+			// For Eg. In Currency-Switcher add on when donation has been made through
+			// different currency other than base currency, in that case for correct
+			//report calculation based on base currency we will need to return donation
+			// base amount and not the converted amount .
+			'type'     => '',
```

Comment:
@emgk @ravinderk should there be a default value for this to prevent the need to always define this?

Reply:
@DevinWalker I do not think there should be any default `type`, because if we will add default type and somebody apply their filter to it then it can break amount unintentionally.

@emgk what do you think

Reply:
@ravinderk Yes.

@DevinWalker Nope, we don't need any default type for this, because keeping it as blank will return correct donation amount same as `_give_payment_total`, there is the very limited use of this key `type` like getting the amount for the report statistics to get the base amount of donation _( if currency switcher is activated )_ and already passed type wherever it was needed.

Reply:
Thanks guys. Makes sense and I'm glad this got merged into the `array()` arg rather than a new one. 👍